src/operations/blas1/IndexMaxMin.hpp

```diff
@@ -94,22 +94,14 @@ PORTBLAS_INLINE void IndexMaxMin<is_max, is_step0, lhs_t, rhs_t>::eval(
   using element_t =
       typename ResolveReturnType<op, rhs_t>::type::value_t::value_t;
 
-#ifndef __ADAPTIVECPP__
   // reduction within the sub_group
   for (index_t i = sg_local_range >> 1; i > 0; i >>= 1) {
-    if (sg_local_id < i) {
-      element_t shfl_val = sycl::shift_group_left(sg, val.get_value(), i);
-      index_t shfl_idx = sycl::shift_group_left(sg, val.get_index(), i);
-      value_t shfl{shfl_idx, shfl_val};
-      val = op::eval(val, shfl);
-    }
+    element_t shfl_val = cl::sycl::shift_group_left(sg, val.get_value(), i);
+    index_t shfl_idx = cl::sycl::shift_group_left(sg, val.get_index(), i);
+    value_t shfl{shfl_idx, shfl_val};
+    val = op::eval(val, shfl);
```

Comment:
A classic sub-group mistake! Nice work identifying it.

* This is a reduction. It might be out of scope for this PR, but using SYCL group reduction operations would seem more natural than implementing it by hand.
* Can `op::eval` have side affects when called with garbage? Does it need to be in an `if`?
* I don't see the point in using `cl` here when `shift_group_left` is SYCL-2020.

Reply:
Thank you for your comments @hjabird.

- I think that implementing `iamax/iamin` kernels using SYCL group reduction would require further investigation. In my opinion, the reason for this implementation is that for these operators the kernel must keep together value and initial index in the vector, you cannot call two different reductions and combine the result. A possible solution would be create a vector of tuple (value,position) and call a group reduction but, as I said, I have no idea about performance impact.
- For how the kernel is implemented and launched I don't think that garbage values `op::eval` read and used by it would ever ends up in the final result. Every thread read at least one _good_ value from the starting vector and only the first thread of each subgroup write out the result.
- We have this technical debt in other places and we plan to eventually update everywhere it in the future.

Reply:
That all sounds reasonable to me. Thanks!